modules/core/variables.tf

```diff
@@ -28,6 +28,18 @@ variable "vault_ami_id" {
   description = "AMI ID for Vault servers"
 }
 
+variable "consul_route53_subdomain" {
+  description = "Subdomain name of Consul API, paired with the VPC base domain as the full domain"
+}
+
+variable "nomad_route53_subdomain" {
+  description = "Subdomain name of Nomad API, paired with the VPC base domain as the full domain"
+}
+
+variable "vault_route53_subdomain" {
+  description = "Subdomain name of Vault API, paired with the VPC base domain as the full domain"
+}
```

Comment:
Think you need to make these variables `defaut = ""` too.

Perhaps add the description that this overrides `xxx_api_domain` for Route53 resource when given a value?